src/redux/books/books.js

```diff
@@ -1,67 +1,60 @@
-import { v4 as uuidv4 } from 'uuid';
+/* eslint-disable camelcase */
+import { createAsyncThunk } from '@reduxjs/toolkit';
 
+const Api = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/3NpzSS6TCHlgPC9vqAsu/books';
+
+// Actions
 const ADD = 'bookstore/books/addBook';
 const REMOVE = 'bookstore/books/removeBook';
+const GET_BOOKS = 'bookstore/books/fetchBooks';
 
-const initialState = {
-  books: [
-    {
-      id: uuidv4(),
-      category: 'action',
-      title: 'Titanic',
-      author: 'Winfred',
-    },
-    {
-      id: uuidv4(),
-      category: 'action',
-      title: 'Titanic',
-      author: 'Winnie',
-    },
-    {
-      id: uuidv4(),
-      category: 'action',
-      title: 'tanic',
-      author: 'Winfred',
-    },
-    {
-      id: uuidv4(),
-      category: 'action',
-      title: 'Titan',
-      author: 'Winfred',
-    },
-  ],
-};
+const initialState = [];
 
+// Add Reducers
 export default function booksReducer(state = initialState, action) {
   switch (action.type) {
-    case ADD: {
-      const newbook = [...state.books, action.payload];
-      return { ...state.books, books: newbook };
-    }
+    case `${ADD}/fulfilled`:
+      return action.payload;
 
-    case REMOVE:
-      return {
-        ...state,
-        books: state.books.filter((book) => book.id !== action.payload.id),
-      };
+    case `${GET_BOOKS}/fulfilled`:
+      return Object.keys(action.payload).map((key) => {
+        const { title, author, category } = action.payload[key][0];
+        return {
+          item_id: key,
+          title,
+          author,
+          category,
+        };
+      });
 
+    case `${REMOVE}/fulfilled`:
+      return state.filter((book) => book.id !== action.payload);
     default:
       return state;
   }
 }
 
-export const addBook = (title, author) => ({
-  type: ADD,
-  payload: {
-    title,
-    author,
-    id: uuidv4(),
-  },
+export const fetchBooks = createAsyncThunk(GET_BOOKS, async () => {
+  const response = await fetch(Api);
+  const data = await response.json();
+  return data;
 });
 
-export const removeBook = (id) => ({
-  type: REMOVE,
-  payload: {
-    id,
-  },
+export const addBook = createAsyncThunk(ADD, async (book) => {
+  await fetch(Api, {
+    method: 'POST',
+    body: JSON.stringify(book),
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+});
+
+export const removeBook = createAsyncThunk(REMOVE, async (id) => {
+  await fetch(`${Api}/${id}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
 });
```

Comment:
:red_circle: You have done well implementing the remove a book via API feature but your front-end view does not update after the promise gets fulfilled. 
:bulb: Maybe what you can do to fix this, Kindly Make sure you return the id or an instance of the book  you have removed then use that in your reducer

```
export const remove = async (id) => {
  await fetch(`${Api}/${id}`, {
    method: 'DELETE',
    headers: {
      'Content-type': 'application/json',
    },
  });
// Keep a reference for later use
 return { id };
};
```